distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -0,0 +1,379 @@
+    private static final char PARSING_CHARACTER='0';
+    // The length of the Node_ID
+    public static final int ID_LENGTH = 128;
+    public static final int MIN_ID = 1;
+    public static final int MAX_ID = 999;
+    public static final String RESOURCE_SEPARATOR="\r";
+    public static final int MIN_PARTS=1;
+    public static final int MAX_PARTS = 999;
+    private final String ACTION_CODE_NOT_FOUND_ERROR_MSG = "Expected ActionType as int number, found not parsable String instead";
+    private final String PAYLOAD_TYPE_NOT_FOUND_ERROR_MSG="Expected PayloadType as int number, found not parsable String instead";
+
+    private SMSPeer actionPeer;
+    private ActionType actionType;
+    private int operationId;
+    private int currentPart;
+    private int totalParts;
+    private PayloadType payloadType;
+    private String payload;
+
+
+    /**
+     * Enum for Action type
+     */
+    public enum ActionType {
+        INVALID(-1),
+        //Codes should be even for Requests, odd for Responses
+        PING(0),
+        PING_ANSWER(1),
+        INVITE(2),
+        INVITE_ANSWER(3),
+        STORE(4),
+        STORE_ANSWER(5),
+        FIND_NODE(6),
+        FIND_NODE_ANSWER(7),
+        FIND_VALUE(8),
+        FIND_VALUE_ANSWER(9);
+
+        private final int code;
+
+        /**
+         * Constructor. Used only by enum.
+         *
+         * @param code the value for code field.
+         */
+        ActionType(int code) {
+            this.code = code;
+        }
+
+        /**
+         * @return the numerical value for the instance
+         */
+        public int getCode() {
+            return code;
+        }
+
+        /**
+         * @return true if this is a Request Action type, false if it is a Response type
+         */
+        public boolean isRequest() {
+            return (code >= 0) && (code % 2 == 0);
+        }
+
+        /**
+         * @return true if this is a Request Action type, false if it is a Response type
+         */
+        public boolean isResponse() {
+            return (code >= 0) && (code % 2 != 0);
+        }
+
+        /**
+         * @param code the code for the Action
+         * @return the ActionType with the corresponding code, or INVALID if an invalid code is passed.
+         */
+        public static ActionType getTypeFromVal(int code) {
+            for (ActionType type : ActionType.values()) {
+                if (type.code == code) return type;
+            }
+            return INVALID;
+        }
+    }
+
+    /**
+     * Enum for type of content attached to the Action, if any
+     */
+    public enum PayloadType {
+
+        INVALID(-1),
+        IGNORED(0),
+        BOOLEAN(1),
+        PEER_ADDRESS(2),
+        NODE_ID(3),
+        RESOURCE(4);
+
+        private final int code;
+
+        /**
+         * Constructor. Used only by enum.
+         *
+         * @param code the value for code field.
+         */
+        PayloadType(int code) {
```

Comment:
@kummy98

Reply:
I was thinking about the same thing, but I couldn't know how to comment it, since by seeing your way to manage enum made me realize that I didn't know enough about enum types.